docs/mathlib-overview.md

```diff
@@ -0,0 +1,60 @@
+
+### Topology
+
+Surprisingly located in `analysis/topology/`, the basic topology files contain an API for topological spaces and continuous maps, as well as the theory of uniform spaces and (in `analysis/`) metric spaces. There is also an API for topological groups and rings (as well as topological monoids and various other things it was appropriate to include), as well as a very general theory of limits (via filters) and a basic API for infinite sums.
+
+### Category theory
+
+Because the categories that mathematicians want to work with are often not sets, there are technical foundational issues here. The category theory library for mathematicians is in `category_theory/` and currently only has a few basic definitions (functors, natural transformations), which have been hard won. Do not confuse this with the `category/` directory in mathlib, which is, for the most part, technical tools for use by computer scientists developing tactics.
+
+### Number theory
+
+There is very little here beyond APIs for naturals, integers and rationals, because a general theory of number fields needs basic results about finite-dimensional vector spaces and Noetherian rings and modules, neither of which are present, and basic analytic number theory will need some basic differential and integral calculus, which is also not present. The delicate result that the power function is Diophantine is proved -- this is a key step in Matiyesevich's resolution of Hilbert's 10th problem. Along the way a basic theory of Pell's equation is developed, involving developing some arithmetic of the ring `Z[sqrt(d)]` for d a non-square.
+
+### Other mathematical theories and objects.
+
+The `order/` directory contains the theories of lattices and complete lattices, boolean algebras, filters and Galois connections, which are not often run into in a mathematics degree, although they are the sorts of things which need to be developed if one is to develop mathematics a la Bourbaki.
+
+
+
+### more computer-sciencey things
+
+`computability/` is stuff like Halting problem. `category/` is monad stuff, `logic` is also more for the CS people. `meta/` and `tactic/` is tactics. `set_theory/` is stuff like ordinals and cardinals, and also a complete model of ZFC in Lean, that is, the definition of, and basic API for, a type called `Set` which satisfies the axioms of ZFC set theory.
+
+The directory `data/equiv/` contains an API for the concept of two objects being in bijection with one another. The directories `data/list/`, `data/set` and `data/digma` contain more advanced APIs for lists, sets and sigma types than are given in core Lean. Also in `data/` is more advanced APIs for booleans, characters, finite sets, multisets (that is, finite sets where elements can appear with multiplicity greater than one) and subtype, amongst many other things which form part of the general framework and which mathematicians may not need to know too much about.
```

Comment:
> mathematicians may not need to know too much about

I think you should be clearer about what this means. This stuff is not mathematically interesting, but you will find it somewhere between useful and essential for doing just about anything else. I think you should be more upfront about the purpose of this article. Specifically, what is your intended audience, and what are you trying to show? Is it the stuff mathematicians *want* or what they *need*? :)